src/backup-metadata/test/e2e/e2e_test.go

```diff
@@ -0,0 +1,59 @@
+package e2e_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os/exec"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"code.cloudfoundry.org/capi-k8s-release/src/backup-metadata/internal/cfmetadata"
+)
+
+var _ = Describe("e2e", func() {
+	It("Assert CF metadata structure", func() {
+		backupName := generateRandomName()
+		createBackup(backupName)
+		cfMetadataJSON := getCFMetadataFromBackupLogs(backupName)
+
+		var metadata cfmetadata.Metadata
+		err := json.Unmarshal([]byte(cfMetadataJSON), &metadata)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(metadata.Totals.Orgs).Should(BeNumerically(">=", 1))
+		Expect(metadata.Totals.Spaces).Should(BeNumerically(">=", 0))
+		Expect(metadata.Totals.Users).Should(BeNumerically(">=", 0))
+		Expect(metadata.Totals.Apps).Should(BeNumerically(">=", 0))
+		Expect(len(metadata.Orgs)).Should(BeNumerically(">=", 1))
+	})
+})
+
+func generateRandomName() string {
+	return "backup-" + random()
+}
+
+func random() string {
+	rand.Seed(time.Now().UnixNano())
+
+	return strconv.Itoa(rand.Intn(1000))
+}
+
+func createBackup(backupName string) {
+	fmt.Printf("Creating velero backup: %s \n", backupName)
+
+	_, err := exec.Command("velero", "backup", "create", "--wait", backupName).Output()
```

Comment:
How does this work exactly, and what does it back up? Is this a "partial" test backup or is there some configuration somewhere I'm missing that configures `velero backup create` to backup CCDB, k8s, etc?

Reply:
Given that `velero` is installed on the cluster. The annotation we have on our pod (https://github.com/cloudfoundry/capi-k8s-release/pull/66/files#diff-d4b5178aed59d7f83f3e3f1b63e2d40cR20-R21) is invoked by velero and the backup-metadata binary is executed. Velero will backup all k8s resources. Regarding CCDB, given that it's more to tanzu postgres or an external DB, this is something we are working with the Tanzu Postgres Team to add a similar annotation which will invoke a custom backup procedure.

Reply:
> Velero will backup all k8s resources and disks

In the cf-for-k8s default case, does this back up CCDB? Is that what this tests?

> Tanzu postgres or an external DB

This is outside the scope of this repo and these tests :)